.github/workflows/fetch-people-profiles.yml

```
@@ -26,8 +26,18 @@ jobs:
         with:
           node-version-file: ${{ env.GITHUB_WORKSPACE }}/.nvmrc
 
-      - run: corepack enable pnpm
-      - run: pnpm add -w gray-matter
+      - name: Enable corepack
+        run: |
+          corepack enable
+
+      - name: Prepare pnpm
+        run: |
+          corepack prepare pnpm@9.15.1 --activate # It doesn't work well with pnpm@latest, therefore we use a specific version as per - https://github.com/pnpm/pnpm/issues/9029
```

```suggestion
      - uses: pnpm/action-setup@v4
```
this action will install pnpm and use the version specified in the package.json, which is better than hardcoding